lib/api/rule.go

```diff
@@ -58,8 +62,12 @@ func init() {
 
 func validateRule(v *validator.Validate, structLevel *validator.StructLevel) {
 	rule := structLevel.CurrentStruct.Interface().(Rule)
-	if rule.ICMPCode != nil && rule.ICMPType == nil {
-		structLevel.ReportError(reflect.ValueOf(rule.ICMPCode), "ICMPCode", "icmpCode", "icmpCodeWithoutType")
+	if rule.ICMP != nil && rule.ICMP.Code != nil && rule.ICMP.Type == nil {
+		structLevel.ReportError(reflect.ValueOf(rule.ICMP.Code), "Code", "code", "icmpCodeWithoutType")
+	}
+
+	if rule.NotICMP != nil && rule.NotICMP.Code != nil && rule.NotICMP.Type == nil {
+		structLevel.ReportError(reflect.ValueOf(rule.NotICMP.Code), "Code", "code", "notIcmpCodeWithoutType")
```

Comment:
I think `notIcmpCodeWithoutType` should just be `icmpCodeWithoutType`.

We can distinguish the two cases by doing `Code` -> `NotICMP.Code`?